packages/x-components/src/x-modules/scroll/components/window-scroll.vue

```diff
@@ -33,135 +33,78 @@
-     * Get the element depends on {@link BaseMainScroll.tag} if is html or body
-     * and listen the event scroll.
+     * Sets the HTML element depending on {@link WindowScroll.tag}, and initialises its events.
      *
      * @internal
      */
     protected initAndListenElement(): void {
-      if (this.tag === 'html') {
-        this.element = document.documentElement;
-        this.elementListener = document;
-      } else {
-        this.element = document.body;
-        this.elementListener = document.body;
-      }
-
-      this.elementListener.addEventListener('scroll', this.throttledStoreScrollData);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-expect-error
+      this.$el = this.tag === 'html' ? document.documentElement : document.body;
+      this.$el.addEventListener('scroll', this.throttledStoreScrollData);
     }
 
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    render(): void {}
+
     /**
-     * Updates main scroll related properties.
-     *
-     * @internal
+     * Cleanup listeners.
      */
-    protected storeScrollData(): void {
-      this.currentPosition = this.element.scrollTop;
-      this.scrollHeight = this.element.scrollHeight;
-      this.clientHeight = this.element.clientHeight;
+    beforeDestroy(): void {
+      this.$el.removeEventListener('scroll', this.throttledStoreScrollData);
     }
 
     /**
-     * Emits the event corresponding passed as parameter when the user has scrolled.
-     *
-     * @param event - Name of event to emit.
-     * @internal
-     */
-    protected emitEvent<Event extends XEvent>(event: Event): void;
-    /**
-     * Emits the event corresponding passed as parameter when the user has scrolled.
+     * Creates the metadata for the events of this component.
      *
-     * @param event - Name of event to emit.
-     * @param payload - Data to send in the event like payload. {@link ScrollDirection | number}.
+     * @returns A {@link WireMetadata} for the events emitted by this component.
      * @internal
      */
-    protected emitEvent<Event extends XEvent>(event: Event, payload: XEventPayload<Event>): void;
-    /**
-     * Emits the event corresponding passed as parameter when the user has scrolled.
-     *
-     * @param event - Name of event to emit.
-     * @param payload - Optional data to send in the event like payload.
-     * {@link ScrollDirection | number}.
-     *
-     * @internal
-     */
-    protected emitEvent<Event extends XEvent>(event: Event, payload?: XEventPayload<Event>): void {
-      this.$x.emit(event, payload as any, { target: this.element, id: this.id });
-    }
-
-    // eslint-disable-next-line @typescript-eslint/no-empty-function
-    render(): void {}
-
-    beforeDestroy(): void {
-      this.elementListener.removeEventListener('scroll', this.throttledStoreScrollData);
+    protected createXEventMetadata(): Partial<WireMetadata> {
+      return { target: this.$el, id: this.id };
```

Comment:
This is the same as the createEventMetadata from the `Scroll` component right? Can we extract to the utils folder?

If not, please add the type `as HTMLElement` to the `$el`.

Reply:
+1 for the type

Reply:
I don't think it makes sense to make any of this. 
The `this.$el` is already typed thanks to the `ScrollMixin` component. In the `Scroll.vue` component, it has to be casted, as it does not use the `ScrollMixin`.

I don't see any value in the utility, appart from other layer of abstraction. If this was more complex code with decisions, loops it might make more sense, but for creating a typed object with 2 properties? Unnecessary FMPOV.